action.yml

```diff
@@ -83,7 +83,27 @@ runs:
             process.exit(1);
           }
 
+    - name: Get condition variables
+      uses: actions/github-script@v7
+      env:
+        CLIENT_PAYLOAD_ARG: ${{ inputs.client_payload }}
+      with:
+        script: |
+          require('${{ github.action_path }}/scripts/getConditionVars.js')(core);
+
+    - name: Download cache artifact
+      id: run-data-artifact
+      uses: actions/download-artifact@v4
+      if: ${{ env.IS_NON_COMMIT_EVENT == 'true' && env.ENABLE_CACHE == 'true' }}
+      with:
+        github-token: ${{ fromJSON(fromJSON(inputs.client_payload)).githubToken || github.token }}
+        repository: ${{ inputs.full_repository }}
+        run-id: ${{ fromJSON(fromJSON(inputs.client_payload)).artifactRunId }}
+        name: output
+        path: code/output
+
     - name: Checkout Pull Request branches history
+      if: ${{ !(env.IS_NON_COMMIT_EVENT == 'true' && env.ENABLE_CACHE == 'true') }}
```

Comment:
since this condition `!(env.IS_NON_COMMIT_EVENT == 'true' && env.ENABLE_CACHE == 'true')`  is repeated 3 times, it may be good to set some variables in script and to use don't it, for example, `SKIP_GIT_CLONE`. 
otherwise if may update it in one place and forget about all the rest places.
WDYT?

Reply:
yeah good idea. changed it